pkg/controller/migplan/storage.go

```diff
@@ -74,27 +82,80 @@ func (r ReconcileMigPlan) ensureStorage(plan *migapi.MigPlan) error {
 	return err
 }
 
+// Delete associated storage.
+func (r ReconcileMigPlan) ensureStorageDeleted(plan *migapi.MigPlan) error {
+	clusters, err := migapi.ListClusters(r)
+	if err != nil {
+		return err
+	}
+	for _, cluster := range clusters {
+		client, err := cluster.GetClient(r)
+		if err != nil {
+			return err
+		}
+		// BSL
+		bsl, err := plan.GetBSL(client)
+		if err != nil {
+			return err
+		}
+		if bsl != nil {
+			err := client.Delete(context.TODO(), bsl)
+			if err != nil && !errors.IsNotFound(err) {
+				return err
+			}
+		}
+		// VSL
+		vsl, err := plan.GetVSL(client)
+		if err != nil {
+			return err
+		}
+		if vsl != nil {
+			err := client.Delete(context.TODO(), vsl)
+			if err != nil && !errors.IsNotFound(err) {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+//
+// PlanStorage
+// Client: The controller client.
+// targetClient: A client for a cluster.
+// plan: A plan resource.
+// storage: A storage resource.
+//
+type PlanStorage struct {
```

Comment:
Represents the association between the plan, storage and a cluster.  And, avoids passing so many (same) parameters to all the functions.